vendor: bump Pebble to 31d5cf45a826

jbowens opened the pull request:
```
31d5cf4 vfs: modify Create to remove and create existing files
61318a0 Fix word repetition
0cae417 internal/manifest: remove redundant `files` field
07228a5 internal/manifest: remove superfluous for-loop
3e0b8e6 pebble: Preserve one previous manifest
5c776d6 db: write manifest before creating WAL during Open
5f4e863 internal/errorfs: allow more granular error injection
1306f1b build: fix mod-update target
```

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/68925)
<!-- Reviewable:end -->

jbowens committed:
vendor: bump Pebble to 31d5cf45a826

31d5cf4 vfs: modify Create to remove and create existing files
61318a0 Fix word repetition
0cae417 internal/manifest: remove redundant `files` field
07228a5 internal/manifest: remove superfluous for-loop
3e0b8e6 pebble: Preserve one previous manifest
5c776d6 db: write manifest before creating WAL during Open
5f4e863 internal/errorfs: allow more granular error injection
1306f1b build: fix mod-update target

Release note: None

jbowens commented:
Superseded by #69114.
